src/main/java/org/elasticsearch/common/io/stream/ByteBufferStreamInput.java

```diff
@@ -72,7 +72,7 @@ public long skip(long n) throws IOException {
 
     @Override
     public void readBytes(byte[] b, int offset, int len) throws IOException {
-        if (buffer.remaining() > len) {
+        if (buffer.remaining() < len) {
             throw new EOFException();
```

Comment:
wtf... now that this class seems to actually be used, should we look at this more in a followup?

For example skip(n) doesn't adhere to the inputstream.skip contract, if the input is negative.

And maybe we should really look at whether inputstream is appropriate for our purposes here. I hate that many InputStream methods are defined with leniency: instead of throwing an exception.